test-context-macros/src/lib.rs

```diff
@@ -135,3 +131,41 @@ fn handle_result(result_name: Ident) -> proc_macro2::TokenStream {
         }
     }
 }
+
+// TODO: Possible bugs when using: type aliases, full and relative paths, generic types in context type
```

Comment:
How could this be addressed?

Honestly this gets out of my knowledge on rust metaprogramming so I'm having a difficult time reviewing it, but I appreciate the contribution and will try to understand it and address it

Reply:
oh alright, I think thats acceptable :D Should we just document that in the README? Then I think we can merge it